user/api.go

```diff
@@ -650,22 +650,24 @@ func (a *Api) UpdateUser(res http.ResponseWriter, req *http.Request, vars map[st
 		if updateUserDetails.EmailVerified != nil {
 			updatedUser.EmailVerified = *updateUserDetails.EmailVerified
 		}
-
+		// Start by updating username on Auth0
+		if a.auth0Client != nil {
+			if usr, err := a.auth0Client.GetUserById(updatedUser.Id); err == nil && usr != nil {
+				if usr.Username == updatedUser.Username {
+					// no need to update with the same value
+					auth0User.Username = nil
+				}
+				if err := a.auth0Client.UpdateUser(updatedUser.Id, auth0User); err != nil {
+					a.sendError(res, http.StatusInternalServerError, STATUS_ERR_UPDATING_USR, log, err)
+					return
+				}
+			}
+		}
+		// And finally update the local db
 		if err := a.Store.UpsertUser(req.Context(), updatedUser); err != nil {
 			a.sendError(res, http.StatusInternalServerError, STATUS_ERR_UPDATING_USR, log, err)
 		} else {
-			if a.auth0Client != nil {
-				if usr, err := a.auth0Client.GetUserById(updatedUser.Id); err == nil && usr != nil {
-					if usr.Username == updatedUser.Username {
-						// no need to update with the same value
-						auth0User.Username = nil
-					}
-					if err := a.auth0Client.UpdateUser(updatedUser.Id, auth0User); err != nil {
-						a.logger.Error("Impossible to update user on Auth0: ", err)
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches to further communicate your intent.  -->
**🎯 suggestion (non-blocking):** ‏I think this log would still be useful even now for the investigations? (and it would be even better to add the user email/user id in the log :star_struck:)

Reply:
Corrected